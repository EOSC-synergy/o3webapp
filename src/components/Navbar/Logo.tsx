import React, { FC } from 'react';
import PropTypes from 'prop-types';
import Link from '@mui/material/Link';
import Grid from '@mui/material/Grid';
import Typography from '@mui/material/Typography';

export const LOGO_SOURCE =
    process.env.NEXT_PUBLIC_LOGO_SRC ?? 'https://o3as.data.kit.edu/img/logos/o3as-logo.svg';
export const LOGO_NAVIGATION = '#page-top';
const LOGO_TEXT = 'O3as: Webapp';

type LogoProps = {
    // TODO: Object typing? see https://mui.com/system/display/
    display?: string | object;
};
/**
 * Returns a logo from O3AS and the text "O3AS Webapp".
 *
 * @param display See https://mui.com/system/display/
 */
const Logo: FC<LogoProps> = ({ display }) => {
    return (
        <Link
            href={LOGO_NAVIGATION}
            sx={{
                flexGrow: 1,
                textDecoration: 'inherit',
                color: 'inherit',
                display: display,
            }}
            noWrap
            data-testid="logo-container"
        >
            <Grid container sx={{ alignItems: 'center', margin: '0.5em', marginLeft: '2em' }}>
                <img
                    data-testid="logo-image"
                    style={{ height: '4em', margin: '0.5em' }}
                    src={LOGO_SOURCE}
                    alt="logo"
                />
                <Typography
                    variant="h6"
                    noWrap
                    data-testid="logo-text"
                    sx={{
                        color: '#C7D3E0',
                        '&:hover': {
                            color: '#fed136',
                        },
                        fontSize: '28px',
                    }}
                >
                    {LOGO_TEXT}
                </Typography>
            </Grid>
        </Link>
    );
};

Logo.propTypes = {
    /** Display style to pass to the logo */
    display: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
};

export default Logo;
