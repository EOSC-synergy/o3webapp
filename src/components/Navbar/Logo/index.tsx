import React from 'react';
import PropTypes from 'prop-types';
import Link from '@mui/material/Link';
import Grid from '@mui/material/Grid';
import Typography from '@mui/material/Typography';

/**
 * The link that provides the image logo
 * @constant {String}
 * @memberof Logo
 * @default "https://o3as.data.kit.edu/img/logos/o3as-logo.svg"
 */
export const logoSrc = 'https://o3as.data.kit.edu/img/logos/o3as-logo.svg';

/**
 * The link to the O3AS Webapp
 * @constant {String}
 * @memberof Logo
 * @default "#page-top"
 */
export const O3ASLink = '#page-top';

type LogoProps = {
    // TODO: Object typing? see https://mui.com/system/display/
    display?: string | Object;
};
/**
 * Returns a logo from O3AS and the text "O3AS Webapp".
 * @component
 * @param display see https://mui.com/system/display/
 * @returns {JSX.Element} JSX Element containing one image and one text wrapped in a Link component
 */
const Logo: React.FC<LogoProps> = ({ display }): JSX.Element => {
    /**
     * Name of the webapp that should be displayed in the logo
     * @constant {String}
     * @default "O3as: Webapp"
     */
    const logoName = 'O3as: Webapp';

    return (
        <Link
            href={O3ASLink}
            sx={{
                flexGrow: 1,
                textDecoration: 'inherit',
                color: 'inherit',
                display: display,
            }}
            noWrap
            data-testid="logo-container"
        >
            <Grid container sx={{ alignItems: 'center', margin: '0.5em', marginLeft: '2em' }}>
                <img
                    data-testid="logo-image"
                    style={{ height: '4em', margin: '0.5em' }}
                    src={logoSrc}
                    alt={'logo'}
                />
                <Typography
                    variant="h6"
                    noWrap
                    data-testid="logo-text"
                    sx={{
                        color: '#C7D3E0',
                        '&:hover': {
                            color: '#fed136',
                        },
                        fontSize: '28px',
                    }}
                >
                    {logoName}
                </Typography>
            </Grid>
        </Link>
    );
};

Logo.propTypes = {
    /**
     * display style to pass to the logo
     */
    display: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
};

export default Logo;
