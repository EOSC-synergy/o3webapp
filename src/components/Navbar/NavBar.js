import * as React from 'react';
import {AppBar, Button, Container, Grid, Toolbar} from '@mui/material';
import Logo from './Logo/Logo';
import EditIcon from '@mui/icons-material/Edit';
import Typography from "@mui/material/Typography";
import { BACKGROUND_BASE_COLOR } from '../../utils/constants';

/**
 * A component for the navigation of the website.
 * @component
 */
function Navbar(props) {

    /**
     * whether the menu (when the navbar collapsed) is currently visible
     */
    const [anchorElNav, setAnchorElNav] = React.useState(null);

    /**
     * Opens the nav menu.
     *
     * @param event the event that triggered this function
     */
    const handleOpenNavMenu = (event) => {
        setAnchorElNav(event.currentTarget);
    };

    /**
     * Closes the nav menu.
     */
    const handleCloseNavMenu = () => {
        setAnchorElNav(null);
    };

    /**
     * When the edit graph button is clicked in collapsed mode,
     * the nav menu will close first before the sidebar will open.
     */
    const handleEditGraphButtonCollapsed = () => {
        handleCloseNavMenu();
        props.openSidebar();
    }

    return (
        <div id="Navbar">
            <AppBar position="static" sx={{ margin: 0, bgcolor: BACKGROUND_BASE_COLOR}} data-testid="Navbar" >
                <Container disableGutters maxWidth={false} sx={{ margin: 0 }} >
                    <Toolbar disableGutters data-testid="Toolbar_Navbar">
                        <Grid
                            container
                            direction="row"
                            justifyContent="center"
                            alignItems="center"
                            key="navbar grid container"
                            sx={{backgroundColor: BACKGROUND_BASE_COLOR}}
                        >
                            <Grid
                                item
                                xs={8}
                                key="logo grid item"
                            >
                                <Logo display='flex' data-testid="navbar-logo-expanded" />
                            </Grid>
                            <Grid
                                item
                                key="edit graph button grid item"
                            >
                                <Button
                                     key={"editGraphButton"}
                                     onClick={props.openSidebar}
                                     sx={{
                                         color: 'white',
                                        '&:hover': {
                                             color: '#fed136',
                                        },
                                     }}
                                >
                                    <Typography>Edit Graph <EditIcon sx={{fontSize: '14px'}} /></Typography>
                                </Button>
                            </Grid>
                        </Grid>
                    </Toolbar>
                </Container>
            </AppBar>
        </div>
    );
}

Navbar.propTypes = {

};

export default Navbar;