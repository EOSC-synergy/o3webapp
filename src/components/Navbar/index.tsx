import React, { FC } from 'react';
import { AppBar, Button, Container, Grid, Toolbar } from '@mui/material';
import Logo from './Logo';
import EditIcon from '@mui/icons-material/Edit';
import Typography from '@mui/material/Typography';
import { BACKGROUND_BASE_COLOR } from 'utils/constants';

type NavBarProps = {
    openSidebar?: () => void;
};
/** A component for the navigation of the website. */
const NavBar: FC<NavBarProps> = ({ openSidebar = () => undefined }) => {
    return (
        <div id="NavBar" style={{ flex: '0 1 auto' }}>
            <AppBar
                position="static"
                sx={{ margin: 0, bgcolor: BACKGROUND_BASE_COLOR }}
                data-testid="NavBar"
            >
                <Container disableGutters maxWidth={false} sx={{ margin: 0 }}>
                    <Toolbar disableGutters data-testid="Toolbar_Navbar">
                        <Grid
                            container
                            direction="row"
                            justifyContent="center"
                            alignItems="center"
                            key="navbar grid container"
                            sx={{ backgroundColor: BACKGROUND_BASE_COLOR }}
                        >
                            <Grid item xs={8} key="logo grid item">
                                <Logo display="flex" data-testid="navbar-logo-expanded" />
                            </Grid>
                            <Grid item key="edit graph button grid item">
                                <Button
                                    key={'editGraphButton'}
                                    onClick={openSidebar}
                                    sx={{
                                        color: 'white',
                                        '&:hover': {
                                            color: '#fed136',
                                        },
                                    }}
                                >
                                    <Typography>
                                        Edit Graph <EditIcon sx={{ fontSize: '14px' }} />
                                    </Typography>
                                </Button>
                            </Grid>
                        </Grid>
                    </Toolbar>
                </Container>
            </AppBar>
        </div>
    );
};

export default NavBar;
