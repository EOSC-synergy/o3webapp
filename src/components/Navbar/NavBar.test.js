import React from 'react';
import ReactDOM from 'react-dom';
import Navbar from './NavBar';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Navbar />, div);
});

// Snapshot test
it('renders correctly', () => {
    const { asFragment } = render(<Navbar />);
    expect(asFragment()).toMatchSnapshot();
});
