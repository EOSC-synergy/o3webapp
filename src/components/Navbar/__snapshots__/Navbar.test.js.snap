// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    id="Navbar"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-18lypu8-MuiPaper-root-MuiAppBar-root"
      data-testid="Navbar"
    >
      <div
        class="MuiContainer-root MuiContainer-disableGutters css-td5xp0-MuiContainer-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular css-1cndkqf-MuiToolbar-root"
          data-testid="Toolbar_Navbar"
        >
          <div
            class="MuiGrid-root MuiGrid-container css-zvblfl-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vok43g-MuiGrid-root"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiTypography-noWrap MuiLink-root MuiLink-underlineAlways css-2grxvj-MuiTypography-root-MuiLink-root"
                data-testid="logo-container"
                href="#page-top"
              >
                <div
                  class="MuiGrid-root MuiGrid-container css-1e0cb12-MuiGrid-root"
                >
                  <img
                    alt="logo"
                    data-testid="logo-image"
                    src="https://o3as.data.kit.edu/img/logos/o3as-logo.svg"
                    style="height: 4em; margin: 0.5em;"
                  />
                  <h6
                    class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-154cv3q-MuiTypography-root"
                    data-testid="logo-text"
                  >
                    O3as: Webapp
                  </h6>
                </div>
              </a>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-p064m1-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  Edit Graph 
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1ej1pre-MuiSvgIcon-root"
                    data-testid="EditIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                    />
                  </svg>
                </p>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
